### DISCUSSION TITLE
interface to "Networked home micro batteries"

### DISCUSSION THREAD

**Discussion by flatmax** [amberelectric/public-api#87] [Ideas]
Anyone interested in interfacing these home networked microbatteries to the Amber Electric system ?

You can see images of it [here](https://hackaday.io/project/182846-networked-home-battery-v2).

The networked batteries are expandable by simply plugging a new one in and connecting it to your home network. When I say "plug a new one in" I literally mean plug a new one into your hous's standard wall plug! Each of my batteries holds around 3 kWh. They output and input around 1 kW.

If you need say 12 kWh and 4 kW i/o, then you can plug in 4 of these networked batteries. If you need say 24 kWh and 8 kW i/o, then plug in 8 of these networked batteries.

The [electronics](https://github.com/flatmax/BatteryController.electronics) and [code](https://github.com/flatmax/BatteryController) are freely available here.

I've had version 1 running reliably for years and about 3 months ago switched to version 2.

It would be great if someone could interface the system to Amber Electric. At the moment it interfaces directly to an Envoy in the fusebox. That is limiting and my friend David suggested that Amber may be interested in this type of battery.

**Comment by madpilot** [comment]
I'm not a electrician but using suicide plugs to connect the inverters to the house is extremely dangerous. Not to mention extremely illegal.

Do you have an Envoy installed? If not you risk electrocuting a linesman during a power outage, as I don't believe (I could be wrong) the micro-inverters are capable of anti-islanding, without the envoy to control them.

I'm all for DIY, but this seems like a bad idea.

**Reply by flatmax** [reply]
Thanks for the feedback madpilot, it is important to think of edge cases and catch bugs. To tell you the truth, most electricians don't have this knowledge - it is not common knowledge.

For two reasons, there is no concern with respect to safety. The two reasons are that the microinverters are standards compliant (identical to being powered by solar panels) and grid linesmen have had to change their safety standards and approaches to fixing lines since generators (such as panels and inverters) have been in use on the grid.

Regarding the connectors and the Envoy; The method of hooking up these inverters to the house is using standard solar CM4 plugs no major issue there. In fact, each inverter is very low power, this method for drawing power from a battery is extremely safe. The Envoy has absolutely no effect - it could be present or absent, it wouldn't matter.

Here is a little more background as to why there is no safety concern (how the standards work for these particular microinverters). These batteries are currently using the IQ7 microinverters. These IQ7 microinverters are not capable of grid forming - this means, that when the grid voltage drops, then the inverters cease to work. The situation would be different if the batteries were using the IQ8 inverters - they are capable of grid forming, however this still poses no safety concern - again that is due to the safety standards which the inverters implent and further the requirements (standards) of isolation circuitry required when gridforming generators are in use.

Regarding the linesmen. Linesmen have changed the way they operate on street lines over the last few decades. The reason for changing the way they operate is due to either non-compliant inverters OR holes in the safety standards which weren't anticipated. Now days, linesmen short all phases either side of the section which they are working on. By shorting the lines either side of the linesmen, no currenty can pass from the grid to the section on which the lineman is operating. As the lines are shorted the voltage is zero - again a very safe situation. If necessary they will also short house lines, but that assumes that there are either non-standards compliant generators behind the house's fuse box or a grid forming inverter is present without compliant isolation circuitry,

As the inverters are standards compliant and hooked up with standard CM4 connectors and linesmen have methods for ensuring their safety, no matter what types of generators are connected to the grid, there is no reason to raise the alarm.

Again, I really appreciate your interest and concern. I would love to hear of anything you think is innovative, fun, and useful from these home networked microbatteries. Over the last few years I have found them fun and very useful - at times these batteries will render houses carbon neutral - even during the night.
